TicketsDemo.EF/CSVReader/TrainRepositoryCSV.cs

```diff
@@ -0,0 +1,147 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using TicketsDemo.Data.Entities;
+using TicketsDemo.Data.Repositories;
+using System.IO;
+using CsvHelper;
+using System.Globalization;
+
+namespace TicketsDemo.EF.CSVReader
+{
+    public class TrainRepositoryCSV : ITrainRepository
+    {
+        private string _trainFile;
+        private string _carriagesFile;
+        private string _placeFile;
+        //public TrainRepositoryCSV(string trainFile = @"D:\1\3 курс\Proga\TicketsDemo\TicketsDemo\Trains.csv",
+        //                          string carriagesFile= @"D:\1\3 курс\Proga\TicketsDemo\TicketsDemo\Carriages.csv",
+        //                          string placeFile = @"D:\1\3 курс\Proga\TicketsDemo\TicketsDemo\Places.csv") 
+        //{
+        //    _trainFile = trainFile;
+        //    _carriagesFile = carriagesFile;
+        //    _placeFile = placeFile;
+        //}
+        public TrainRepositoryCSV()
+        {
+            _trainFile = AppDomain.CurrentDomain.BaseDirectory + "Trains.csv";
+            _carriagesFile = AppDomain.CurrentDomain.BaseDirectory + "Carriages.csv";
+            _placeFile = AppDomain.CurrentDomain.BaseDirectory + "Places.csv";
+        }
+        #region ITrainRepository Members
+        public List<Train> GetAllTrains()
+        {
+            List<Train> Trains;
+            List<Carriage> Carriages;
+            List<Place> Places;
+            using (StreamReader streamReader = new StreamReader(_trainFile))
+            {
+                using (CsvReader csvReader = new CsvReader(streamReader, System.Globalization.CultureInfo.CurrentCulture))
```

Comment:
extract CsvReader initialization into separatae private function to avoid excessive copy-paste
something like

CsvReader MakeReader(StreamReader fromReader)

Reply:
almost there.

now ensure all maps registrations are moved in to reader creation method AND ensure it's disposal 
(instead of var reader = Reader(...) do using(var reader = Reader(...)) { ... }